src/core/createFrameResponse.ts

```diff
@@ -0,0 +1,58 @@
+import { NextResponse } from 'next/server';
+import { Button, FrameMetadata } from './sharedTypes';
+
+/**
+ * Creates a response with a new valid frame to be rendered.
+ *
+ * @param buttons: The buttons to use for the frame.
+ * @param image: The image to use for the frame.
+ * @param post_url: The URL to post the frame to.
+ * @param refresh_period: The refresh period for the image used.
+ * @returns An HTML response containing metadata for the frame.
+ */
+function createFrameResponse({
+  buttons,
+  image,
+  post_url,
+  refresh_period,
+}: FrameMetadata): NextResponse {
+  const imageHtml = image ? `<meta property="fc:frame:image" content="${image}" />` : '';
+  // Ensure only up to 4 buttons are processed
+  let buttonsHtml = '';
+  if (buttons) {
+    const validButtons = buttons
+      .slice(0, 4)
+      .filter((button): button is Button => button !== undefined);
+    buttonsHtml = validButtons
+      .map((button, index) => {
+        let buttonHtml = `<meta property="fc:frame:button:${index + 1}" content="${button.label}" />`;
+        if (button.action) {
+          buttonHtml += `\n<meta property="fc:frame:button:${index + 1}:action" content="${button.action}" />`;
+        }
+        return buttonHtml;
+      })
+      .join('\n');
+  }
+  const postUrlHtml = post_url ? `<meta property="fc:frame:post_url" content="${post_url}" />` : '';
```

Comment:
We don't need to check `post_url` the type should already check that, I believe.

Reply:
The type does check that, though I think the type doesn't fully conform to the spec. The only required properties are `fc:frame`, `fc:frame:image`, and `og:image`. (and our function doesn't itself handle the og image, see #44)

Reply:
The OG story is tricky, as Next has their own way to handle OG. @cnasc let's keep that conversation in the #44 issue. But is not as easy to solve. Would love your thoughts on that.